tests/unit/test_manifest_auth.py

```diff
@@ -71,23 +71,15 @@ def test_auth_oidc(oidc_secret, valid_spec):
     secret = manifest["secret"]
     assert any(
         [
-            container["name"] == "authentication-plugin"
+            container["name"] == "oauth2-proxy"
             for container in manifest["statefulset"]["spec"]["template"]["spec"][
                 "containers"
             ]
         ]
     )
-    assert any(
-        [
-            container["name"] == "authorization-plugin"
-            for container in manifest["statefulset"]["spec"]["template"]["spec"][
-                "containers"
-            ]
-        ]
-    )
-    assert "authenticationPluginCookieSecret" in secret["data"].keys()
+    assert "oauth2ProxyCookieSecret" in secret["data"].keys()
     auth_container = manifest["statefulset"]["spec"]["template"]["spec"]["containers"][
-        3
+        1
```

Comment:
is there really no other way to do this? :)

Reply:
The problem is that for the purpose of patching the resources resulting from the template rendering, we have to assume a certain order of the containers in the pod. In a first implementation - for exactly that reason - I actually treated each container as a separate resource which was addressable by name and only added them to the pods container array before applying the statefulset to the cluster. However, this was really confusing, because you're patching the proper kubernetes resources everywhere else. This is definitely something that has to be documented better.
So failing tests if the order of containers in the pod is wrong is actually a good thing.

Reply:
Your comment actually made me double check: 
https://github.com/json-patch/json-patch2/issues/18